samples/string_interning.cpp

```diff
@@ -0,0 +1,18 @@
+#include <vector>
+#include <string>
```

Comment:
Do we need that?

Reply:
Because I'm using strlen, strncpy and vector.

Reply:
Oh yes, my bad for the `<string>`. Btw, I wanted to only mention line 2. But Github shows more than one lines and I don't know how to only show a single line. So, in this case it seems like I also referenced line 1 (i.e. `<vector>`) but it was not intended.
I was very confused on the start with all this.

Reply:
Ok good to know